dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/CollectionRepositoryImpl.java

```diff
@@ -138,6 +189,64 @@ public long count() {
+        new StringBuilder(
+            "SELECT c.uuid c_uuid, c.refid c_refId, c.label c_label, c.description c_description,"
+                + " c.entity_type c_entityType,"
+                + " file.uuid f_uuid, file.filename f_filename, file.mimetype f_mimeType, file.size_in_bytes f_sizeInBytes, file.uri f_uri, file.iiif_base_url f_iiifBaseUrl"
+                + " FROM collections as c"
+                + " LEFT JOIN fileresources_image as file on c.previewfileresource = file.uuid"
+                + " LEFT JOIN LATERAL jsonb_object_keys(c.label) l(keys) on c.label is not null"
+                + " LEFT JOIN LATERAL jsonb_object_keys(c.description) n(keys) on c.description is not null"
+                + " WHERE (c.label->>l.keys ilike '%' || :searchTerm || '%'"
+                + " OR c.description->>n.keys ilike '%' || :searchTerm || '%')");
+    addPageRequestParams(searchPageRequest, query);
+
+    List<Collection> result =
+        new ArrayList(
+            dbi.withHandle(
+                h ->
+                    h.createQuery(query.toString())
+                        .bind("searchTerm", searchPageRequest.getQuery())
+                        .registerRowMapper(BeanMapper.factory(CollectionImpl.class, "c"))
+                        .registerRowMapper(BeanMapper.factory(ImageFileResourceImpl.class, "f"))
+                        .reduceRows(
+                            new LinkedHashMap<UUID, Collection>(),
+                            (map, rowView) -> {
+                              Collection collection =
+                                  map.computeIfAbsent(
+                                      rowView.getColumn("c_uuid", UUID.class),
+                                      uuid -> rowView.getRow(CollectionImpl.class));
+                              if (rowView.getColumn("f_uuid", String.class) != null) {
+                                collection.setPreviewImage(
+                                    rowView.getRow(ImageFileResourceImpl.class));
+                              }
+                              return map;
+                            })
+                        .values()));
+
+    String countQuery =
+        "SELECT count(*) FROM collections as c"
+            + " LEFT JOIN LATERAL jsonb_object_keys(c.label) l(keys) on c.label is not null"
+            + " LEFT JOIN LATERAL jsonb_object_keys(c.description) n(keys) on c.description is not null"
+            + " WHERE (c.label->>l.keys ilike '%' || :searchTerm || '%'"
+            + " OR c.description->>n.keys ilike '%' || :searchTerm || '%')";
+    long total =
+        dbi.withHandle(
+            h ->
+                h.createQuery(countQuery)
+                    .bind("searchTerm", searchPageRequest.getQuery())
+                    .mapTo(Long.class)
+                    .findOne()
+                    .get());
+
+    SearchPageResponse pageResponse = new SearchPageResponseImpl(result, searchPageRequest, total);
+    return pageResponse;
```

Comment:
Oder auch einfach:

```suggestion
    return new SearchPageResponseImpl(result, searchPageRequest, total);
```

😅

Reply:
Hab ich entsprechend eingebaut :+1: